File: src/main/java/sprykerFileCreator/model/command/UpdateDocBlockCommand.java
package pav.sprykerFileCreator.model.command;

import com.intellij.openapi.project.Project;
import com.jetbrains.php.lang.documentation.phpdoc.psi.PhpDocComment;
import com.jetbrains.php.lang.psi.PhpPsiElementFactory;
import com.jetbrains.php.lang.psi.elements.PhpClass;
import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.generator.DocBlockGeneratorInterface;
import pav.sprykerFileCreator.model.matcher.ClassTypeMatcherInterface;
import pav.sprykerFileCreator.model.renderer.PhpClassRendererInterface;
import pav.sprykerFileCreator.model.renderer.dto.DocBlockItem;

import java.util.List;

public class UpdateDocBlockCommand {

    private ClassTypeMatcherInterface classTypeMatcher;
    private final DocBlockGeneratorInterface docBlockGenerator;
    private final PhpClassRendererInterface phpRenderer;

    public UpdateDocBlockCommand(
            ClassTypeMatcherInterface classTypeMatcher,
            DocBlockGeneratorInterface docBlockGenerator,
            PhpClassRendererInterface phpRenderer
    ) {
        this.classTypeMatcher = classTypeMatcher;
        this.docBlockGenerator = docBlockGenerator;
        this.phpRenderer = phpRenderer;
    }

    public void updateDocBlock(PhpClass phpClass, Project project) throws Exception {
        String fqClassName = phpClass.getFQN();

        ClassDefinitionInterface classDefinition = this.classTypeMatcher.matchClassType(fqClassName);
        String bundleName = this.classTypeMatcher.matchBundleNameFromFQName(classDefinition, fqClassName);

        PhpDocComment oldComment = phpClass.getDocComment();


        List<DocBlockItem> docBlockItems = docBlockGenerator.getDocBlockItems(classDefinition.getDocBlockClasses(), bundleName);

        PhpDocComment comment = PhpPsiElementFactory.createFromText(
                project,
                PhpDocComment.class,
                this.phpRenderer.renderDocBlock(docBlockItems)
        );

        if (comment == null) {
            return;
        }

        if (oldComment == null) {
            phpClass.addBefore(comment, phpClass.getFirstChild());
        } else {
            oldComment.replace(comment);
        }

    }
}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/command/UpdateDocBlockCommand.kt
package com.github.tobi812.sprykerplugin.models.command

import com.github.tobi812.sprykerplugin.models.definitions.ClassDefinitionInterface
import com.github.tobi812.sprykerplugin.models.generator.DocBlockGeneratorInterface
import com.github.tobi812.sprykerplugin.models.matcher.ClassTypeMatcherInterface
import com.github.tobi812.sprykerplugin.models.renderer.PhpClassRendererInterface
import com.github.tobi812.sprykerplugin.models.renderer.dto.DocBlockItem
import com.intellij.openapi.components.service
import com.intellij.openapi.project.Project
import com.jetbrains.php.lang.documentation.phpdoc.psi.PhpDocComment
import com.jetbrains.php.lang.psi.PhpPsiElementFactory
import com.jetbrains.php.lang.psi.elements.PhpClass
import java.lang.Exception

class UpdateDocBlockCommand() {
    @Throws(Exception::class)
    fun updateDocBlock(phpClass: PhpClass, project: Project) {
        val fqClassName = phpClass.fqn
        val classTypeMatcher = project.service<ClassTypeMatcherInterface>()
        val classDefinition: ClassDefinitionInterface = classTypeMatcher.matchClassType(fqClassName)
            ?: throw Exception("Class definition was not found: $fqClassName")

        val bundleName: String = classTypeMatcher.matchBundleNameFromFQName(classDefinition, fqClassName)
        val oldComment = phpClass.docComment
        val docBlockGenerator = project.service<DocBlockGeneratorInterface>()
        val docBlockItems: List<DocBlockItem> = docBlockGenerator.getDocBlockItems(classDefinition.docBlockClasses, bundleName)
        val comment = PhpPsiElementFactory.createFromText(
                project,
                PhpDocComment::class.java,
                project.service<PhpClassRendererInterface>().renderDocBlock(docBlockItems)
        ) ?: return

        if (oldComment == null) {
            phpClass.addBefore(comment, phpClass.firstChild)
        } else {
            oldComment.replace(comment)
        }
    }
}
